foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/generator/MachineLearningPipelineGenerator.java

```diff
@@ -95,18 +94,15 @@ private boolean enableAutoTrain(MachineLearningPipeline pipeline) {
 
     private boolean pipelineContainsOnnxPostAction(MachineLearningPipeline pipeline) {
         boolean containsOnnx = false;
-
-        // this surrounding if-check was added to prevent piplines that use sagemaker training from
-        // throwing a null pointer exception when detecting if Onnx conversion code is needed to be generated
-        if (pipeline.getTrainingStep() != null) {
-            for (PostAction postAction : pipeline.getTrainingStep().getPostActions()) {
-                if (PipelineUtils.forOnnxModelConversion(postAction)) {
-                    containsOnnx = true;
-                    break;
-                }
+        
+        for (PostAction postAction : pipeline.getTrainingStep().getPostActions()) {
```

Comment:
A: I know the comment said it was for SageMaker pipelines, but the `getTrainingStep` method's contract is to return `null` if there is no training step.  So I think keeping the `null` check/guard here makes sense instead of just relying on program flow to only route pipelines with training steps to this method.